File: tests/source/foo.c
float foo(float a, float b) {
  float x;
  float y;
  if (a < b) {
    float z = 2.0;
    x = b - a;
    y = z + a;
  } else {
    float z = 3.0;
    x = a - b;
    y = z + b;
  }
  return x + y;
}

float main() {
  float x = foo(foo(4.0, 6.0), 3.0);
  return 0;
}

File: tests/actual/foo.ml
[@@@warning "-26"]
[@@@warning "-27"]
[@@@warning "-32"]
[@@@warning "-69"]

open Core

let rec set_at_index (lst : 'a list) (index : int) (value : 'a) : 'a list =
  match lst with
  | [] -> failwith "Index out of bounds"
  | hd :: tl ->
      if index = 0 then value :: tl else hd :: set_at_index tl (index - 1) value

let foo (a : float) (b : float) : float =
  let y, x =
    if Float.( < ) a b then
      let z : float = 2. in
      let x = Float.( - ) b a in
      let y = Float.( + ) z a in
      (y, x)
    else
      let z : float = 3. in
      let x = Float.( - ) a b in
      let y = Float.( + ) z b in
      (y, x)
  in
  Float.( + ) x y

let () =
  let x : float = foo (foo 4. 6.) 3. in
  exit 0
